### PR TITLE
feat: added depositAll

### PR DESCRIPTION
**Pull request by Macarse**

### CODE REVIEW

File: contracts/Vault.vy [hunk 1]
```diff
@@ -302,11 +302,15 @@ def _issueSharesForAmount(_to: address, _amount: uint256) -> uint256:
...
 
-@external
-def deposit(_amount: uint256) -> uint256:
+@internal
+def _deposit(_sender: address, _amount: uint256) -> uint256:
     assert not self.emergencyShutdown  # Deposits are locked out
     assert self._totalAssets() + _amount <= self.depositLimit  # Max deposit reached
 
+    # Depositer should have funds
+    assert self.token.balanceOf(_sender) > 0
```
Comment (fubuloubu) [inline contracts/Vault.vy:311]: Cheaper, no external call required. Invariant that balance if greater than 0 is already demonstrated below by success of `transferFrom` (asserting that `_amount > 0` here ensures that we are always transfering >0 balance below)
```suggestion
    assert _amount > 0
```

File: contracts/Vault.vy [hunk 2]
```diff
@@ -302,11 +302,15 @@ def _issueSharesForAmount(_to: address, _amount: uint256) -> uint256:
...
+    assert self.token.balanceOf(_sender) >= _amount
```
Comment (fubuloubu) [inline contracts/Vault.vy:312]: This is inherently true if `self.token.transferFrom(_sender, self, _amount)` succeeds (so, not necessary)

**Review by fubuloubu — APPROVED** [review]
Minor changes, looks good.

Should add `withdrawAll()` in a separate PR?

File: contracts/Vault.vy [hunk 3]
```diff
@@ -302,11 +302,15 @@ def _issueSharesForAmount(_to: address, _amount: uint256) -> uint256:
...
 
 
-@external
-def deposit(_amount: uint256) -> uint256:
+@internal
+def _deposit(_sender: address, _amount: uint256) -> uint256:
     assert not self.emergencyShutdown  # Deposits are locked out
     assert self._totalAssets() + _amount <= self.depositLimit  # Max deposit reached
 
+    # Depositer should have funds
```
Comment (fubuloubu) [inline contracts/Vault.vy:310]:
```suggestion
    # Ensure we are depositing something
```

Comment (Macarse) [reply contracts/Vault.vy]: fixed

**Pull request merged**